src/io/matpower.jl

```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+
+    # Print the header information  
+    println(io, "%% MATPOWER Case Format : Version 2")
+    println(io, "mpc.version = '2';")
+    println(io)
+    println(io, "%%-----  Power Flow Data  -----%%")
+    println(io, "%% system MVA base")
+    println(io, "mpc.baseMVA = ", mvabase, ";")
+    println(io, "function mpc = ", data["name"])
+        
+    # Print the bus data
+    println(io, "%% bus data")
+    println(io, "%    bus_i    type    Pd    Qd    Gs    Bs    area    Vm    Va    baseKV    zone    Vmax    Vmin")
+    println(io, "mpc.bus = [")
+    for (idx,bus) in sort(buses)
+        println(io, "\t", bus["index"], 
+                    "\t", bus["bus_type"], 
+                    "\t", (pd[bus["index"]]*mvabase), 
+                    "\t", (qd[bus["index"]]*mvabase), 
+                    "\t", (gs[bus["index"]]*mvabase), 
+                    "\t", (bs[bus["index"]]*mvabase), 
+                    "\t", bus["area"], 
+                    "\t", bus["vm"], 
+                    "\t", rad2deg(bus["va"]), 
+                    "\t", bus["base_kv"], 
+                    "\t", bus["zone"], 
+                    "\t", bus["vmax"], 
+                    "\t", bus["vmin"],
+                    "\t", (haskey(bus,"lam_p") ? bus["lam_p"] : ""),   
+                    "\t", (haskey(bus,"lam_q") ? bus["lam_q"] : ""),   
+                    "\t", (haskey(bus,"mu_vmax") ? bus["mu_vmax"] : ""),   
+                    "\t", (haskey(bus,"mu_vmin") ? bus["mu_vmin"] : ""),   
+                    )
+    end  
+    println(io, "];")
+    println(io)  
+    
+    # Print the bus names
+    if haskey(collect(values(buses))[1], "bus_name")
+        println(io, "%% bus names")
+        println(io, "mpc.bus_name = {")
+        for (idx,bus) in sort(buses)
+            println(io, "\t'", bus["bus_name"], "'") 
+        end    
+        println(io, "};")
+        println(io)
+    end
+    
+    # Print the generator data
+    println(io, "%% generator data")
+    println(io, "%    bus    Pg    Qg    Qmax    Qmin    Vg    mBase    status    Pmax    Pmin    Pc1    Pc2    Qc1min    Qc1max    Qc2min    Qc2max    ramp_agc    ramp_10    ramp_30    ramp_q    apf")
+    println(io, "mpc.gen = [")
+    i = 1
+    for (idx,gen) in sort(generators)
+        if idx != gen["index"]
+            warn(LOGGER, "The index of the generator does not match the matpower assigned index. Any data that uses generator indexes for reference is corrupted.");           
+        end  
+        println(io, "\t", gen["gen_bus"], 
+                    "\t", (gen["pg"]*mvabase),
+                    "\t", (gen["qg"]*mvabase),
+                    "\t", (gen["qmax"]*mvabase),
+                    "\t", (gen["qmin"]*mvabase),
+                    "\t", gen["vg"],
+                    "\t", gen["mbase"],
+                    "\t", gen["gen_status"],
+                    "\t", (gen["pmax"]*mvabase),
+                    "\t", (gen["pmin"]*mvabase),
+                    "\t", gen["pc1"],
```

Comment:
By the technical Matpower spec I believe this is where the `haskey` checks should start.